src/server/api/lib/campaign.ts

```diff
@@ -230,24 +231,32 @@ export const copyCampaign = async (options: CopyCampaignOptions) => {
         [count, userId, campaignId]
       );
 
-      // Copy Messaging Service OR use active one
-      const messagingServices = await r
-        .knex("messaging_service")
-        .where({
-          organization_id: newCampaign.organization_id,
-          active: true
-        })
-        .orderByRaw(`is_default desc nulls last`);
-
-      if (messagingServices.length === 0) {
-        throw new Error("No active messaging services found");
+      if (template) {
+        await trx("all_campaign")
+          .update({ is_template: true })
+          .where({ id: newCampaign.id });
       }
 
-      await trx("campaign")
-        .update({
-          messaging_service_sid: messagingServices[0].messaging_service_sid
-        })
-        .where({ id: newCampaign.id });
+      if (!template) {
```

Comment:
Nitpick: should this just be an else block?

Reply:
I think not. The if is to check if we should update the `is_template` variable. And the `!template` check is to add a messaging service to it. Keeping it separate "blocks" of work.